source/OptimaJet.DWKit.StarterApplication/Services/Workflow/WorkflowProductService.cs

```diff
@@ -133,29 +128,43 @@ public async Task ProductProcessChangedAsync(ProductProcessChangedArgs args)
                  .Include(p => p.Project)
                  .ThenInclude(pr => pr.Owner)
                  .FirstOrDefaultAsync();
+
             if (product == null)
             {
                 Log.Error($"Could find Product for ProcessId={args.ProcessId}");
                 return;
             }
 
-            await RemoveTasksByProductId(product.Id);
+            var tasks = await ReassignUserTasksForProduct(product);
+
+            await SendNotifications(tasks, product, args);
+
 
+            // Clear the WorkflowComment
+            if (!String.IsNullOrWhiteSpace(product.WorkflowComment))
+            {
+                // Clear the comment
+                product.WorkflowComment = "";
+                await ProductRepository.UpdateAsync(product);
+            }
+        }
+
+        private async Task SendNotifications(List<UserTask> tasks, Product product, ProductProcessChangedArgs args)
```

Comment:
I am really confused by the `SendNotifications` function.  Why is is running the `Runtime.GetAllActorsForDirectCommandTransitions` again?  

It should be sending out a Notification for each input task (which has a user associated with it).
It doesn't need the args.

Reply:
Cause I've been a hot mess the past couple days. Fix incoming!